gen/epilogue.jl

```diff
@@ -5,10 +5,12 @@ const GDAL_DATA = Ref{String}()
 const PROJ_LIB = Ref{String}()
 
 function __init__()
+    # Cleanup at exit
+    atexit(gdaldestroy)
```

Comment:
Does this work if the user doesn't call `gdalallregister()`?

Reply:
I guess that if you want to do something with GDAL you need at least one driver to be loaded.
Do you know a use case (in Julia, C or C++) where one user load only one or few drivers ?
I could test it.

Reply:
> Do you know a use case (in Julia, C or C++) where one user load only one or few drivers ?

There might be cases, e.g. https://trac.osgeo.org/gdal/ticket/6543 where they're not doing any I/O, and just using the spatial projection functionality.

Outside of that, I seldom see people load drivers individually over calling `gdalallregister()`. For reference: in https://gdal.org/tutorials/raster_api_tut.html#opening-the-file, it's said that "If for some applications it is necessary to limit the set of drivers it may be helpful to review the code from [gdalallregister.cpp](https://github.com/OSGeo/gdal/blob/master/gdal/frmts/gdalallregister.cpp). Python automatically calls GDALAllRegister() when the gdal module is imported."

Reply:
> Does this work if the user doesn't call `gdalallregister()`?

I've tested it locally and it seems to work fine. Feel free to resolve this comment thread, thanks!

Reply:
I have browsed [QGIS/src/crssync/main.cpp](https://github.com/qgis/QGIS/blob/master/src/crssync/main.cpp) and from there:
- `QgsApplication::init` in [QGIS/src/core/qgsapplication.cpp](https://github.com/qgis/QGIS/blob/master/src/core/qgsapplication.cpp)
-  `QgsCoordinateReferenceSystem::syncDatabase()` [QGIS/src/core/proj/qgscoordinatereferencesystem.cpp](https://github.com/qgis/QGIS/blob/master/src/core/proj/qgscoordinatereferencesystem.cpp)

without being able to find any GDAL functions to reproduce it with GDAL.jl.
It seems more deeply embedded. Difficult to derive a test.

Reply:
I'm quite convinced from the bug reports in https://github.com/OSGeo/gdal/blob/fec15b146f8a750c23c5e765cac12ed5fc9c2b85/gdal/gcore/gdaldllmain.cpp#L92-L99 that `gdaldestroy()` is supposed to be callable without needing to call `gdalallregister()` first.